components/webcompat_reporter/browser/webcompat_reporter_service.cc

```diff
@@ -142,6 +146,59 @@ void ProcessContactInfo(
           : "");
 }
 
+std::vector<std::string> GetInstalledComponentIds(
+    webcompat_reporter::WebcompatReporterService::Delegate const*
+        service_delegate) {
+  std::vector<std::string> component_ids;
+  if (!service_delegate) {
+    return component_ids;
+  }
+
+  const auto components = service_delegate->GetComponentInfos();
+  component_ids.resize(components.size());
+
+  std::transform(components.begin(), components.end(), component_ids.begin(),
+                 [](auto c) { return c.id; });
+
+  return component_ids;
+}
+
+using WebcompatCategory = webcompat_reporter::mojom::WebcompatCategory;
+
+constexpr char kHideCookieNoticeCategorForComponentId[] =
+    "cdbbhgbmjhfnhnmgeddbliobbofkgdhe";
+constexpr char kHideNewsletterCategorForComponentId[] =
+    "kdddfellohomdnfkdhombbddhojklibj";
+constexpr char kHideSocialCategorForComponentId[] =
+    "nbkknaieglghmocpollinelcggiehfco";
+constexpr char kHideChatCategorForComponentId[] =
+    "cjoooeeofnfjohnalnghhmdlalopplja";
```

Comment:
Minor typo on constant name (missing `y` on category 🙂):
```suggestion
constexpr char kHideCookieNoticeCategoryForComponentId[] =
    "cdbbhgbmjhfnhnmgeddbliobbofkgdhe";
constexpr char kHideNewsletterCategoryForComponentId[] =
    "kdddfellohomdnfkdhombbddhojklibj";
constexpr char kHideSocialCategoryForComponentId[] =
    "nbkknaieglghmocpollinelcggiehfco";
constexpr char kHideChatCategoryForComponentId[] =
    "cjoooeeofnfjohnalnghhmdlalopplja";
```